app/models/channels.js

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export class Channels {
+    static schema = {
+        name: 'Channels',
+        primaryKey: 'id',
+        properties: {
+            id: 'string',
+            createAt: 'int',
+            updateAt: 'int',
+            deleteAt: {type: 'int', default: 0},
+            team: 'Teams',
+            type: {type: 'string', indexed: true},
+            displayName: 'string?',
```

Comment:
Why make the display name, header, and purpose optional? Shouldn't get just default to the empty string?

Reply:
empty strings takes more space than just null, right?

Reply:
I don't know how Realm stores the data internally, but it's definitely small enough that it wouldn't matter for us. I'd prefer to handle it like Go where the "zero" value for a string is the empty string

Reply:
i think every byte counts on a mobile device

Reply:
I think that's a bit overkill, since I'm fairly confident that the empty string and null end up taking the same amount of memory. Even if there is a small memory savings here, I don't think it's worth the potential errors caused by these fields suddenly becoming nullable when they've always had a value in the past

Reply:
We discussed this 1-1, and we decided to leave it as is since it won't really change much in the long run because the server won't usually send down null for strings and the schema being nullable doesn't affect JS's weak type safety